cpp/src/rolling/jit/code/kernel.cpp

```diff
@@ -32,27 +32,134 @@ const char* kernel =
 R"***(
 #include "operation.h"
 
-template <typename WindowType>
-cudf::size_type get_window(WindowType window, cudf::size_type index) { return 0; }
+template<class T>
+T minimum(T a, T b){
```

Comment:
Is there a reason we're not using `return (b < a) ? b : a;` and/or `std::min`?

https://en.cppreference.com/w/cpp/algorithm/min

Reply:
The code in this code string is to be `jitify`'ed as a kernel so we can't use the `STL` functions. In principle `thrust` can be used here to replace it, yet `jitify` doesn't fully support `thrust` ... What's the difference b/w `return (b < a) ? b : a;` and `return a < b ? a : b;`

Reply:
Oh, sorry. I missed where this was in a string.

`(b < a) ? b : a`: if `a == b`, then `a`
`(a < b) ? a : b`: if `a == b`, then `b`

For consistency sake, let's go with the std's version.

Reply:
The reason this is important in the real world is that types can have overloaded operators. Those operators could introduce strange behavior, where a and b are different, even if they are not lexicographical different. In theory, this shouldn't make a difference in most cases. But in the cases it _does_ make a difference, it's a very obscure bug to try and track down, and could waste a lot of someone's time.